src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests5.cs

```diff
@@ -680,59 +1346,401 @@ class C {
+                );
+        }
+
+        [Fact]
+        public void ExtendedPropertyPatterns_ObsoleteAccessor()
+        {
+            var program = @"
+using System;
+class C
+{
+    public void M1(C1 c1)
+    {
+        _ = c1 is { Prop1.Prop2: null };
+    }
+}
+
+class C1
+{
+    public C2 Prop1
+    {
+        [ObsoleteAttribute(""error Prop1"", true)]
+        get;
+        set;
+    }
+}
+class C2
+{
+    public object Prop2
+    {
+        get;
+        [ObsoleteAttribute(""error Prop2"", true)]
+        set;
+    }
+}
+";
+            var compilation = CreateCompilation(program, parseOptions: TestOptions.RegularWithExtendedPropertyPatterns);
+            compilation.VerifyEmitDiagnostics(
+                // (7,21): error CS0619: 'C1.Prop1.get' is obsolete: 'error Prop1'
+                //         _ = c1 is { Prop1.Prop2: null };
+                Diagnostic(ErrorCode.ERR_DeprecatedSymbolStr, "Prop1.Prop2").WithArguments("C1.Prop1.get", "error Prop1").WithLocation(7, 21)
+                );
+        }
+
+        [Fact]
+        public void ExtendedPropertyPatterns_InaccessibleProperty()
+        {
+            var program = @"
+using System;
+class C
+{
+    public void M1(C1 c1)
+    {
+        _ = c1 is { Prop1.Prop2: null };
+    }
+}
+
+class C1
+{
+    private C2 Prop1 { get; set; }
+}
+class C2
+{
+    private object Prop2 { get; set; }
+}
+";
+            var compilation = CreateCompilation(program, parseOptions: TestOptions.RegularWithExtendedPropertyPatterns);
+            compilation.VerifyEmitDiagnostics(
+                // (7,21): error CS0122: 'C1.Prop1' is inaccessible due to its protection level
+                //         _ = c1 is { Prop1.Prop2: null };
+                Diagnostic(ErrorCode.ERR_BadAccess, "Prop1").WithArguments("C1.Prop1").WithLocation(7, 21)
+                );
+        }
+
+        [Fact]
+        public void ExtendedPropertyPatterns_ExpressionTree()
+        {
+            var program = @"
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+class C
+{
+    public void M1(C1 c1)
+    {
+        Expression<Func<C1, bool>> f = (c1) => c1 is { Prop1.Prop2: null };
+    }
+}
+
+class C1
+{
+    public C2 Prop1 { get; set; }
+}
+class C2
+{
+    public object Prop2 { get; set; }
+}
+";
+            var compilation = CreateCompilation(program, parseOptions: TestOptions.RegularWithExtendedPropertyPatterns);
+            compilation.VerifyEmitDiagnostics(
+                // (9,48): error CS8122: An expression tree may not contain an 'is' pattern-matching operator.
+                //         Expression<Func<C1, bool>> f = (c1) => c1 is { Prop1.Prop2: null };
+                Diagnostic(ErrorCode.ERR_ExpressionTreeContainsIsMatch, "c1 is { Prop1.Prop2: null }").WithLocation(9, 48)
+                );
+        }
+
+        public class FlowAnalysisTests : FlowTestBase
+        {
+            [Fact]
+            public void RegionInIsPattern01()
```

Comment:
>RegionInIsPattern01 [](http://example.com/codeflow?start=24&length=19)

Is this testing a new scenario? Should this test be moved to another file?

Reply:
An IDE test crashed in an earlier iteration of this PR. I created a compiler-level repro. It follows many other flow analysis tests, but I prefered to keep it together with the rest of the feature.